pinot-common/src/main/java/com/linkedin/pinot/common/utils/PeriodicTask.java

```diff
@@ -15,74 +15,34 @@
  */
 package com.linkedin.pinot.common.utils;
 
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 
 public abstract class PeriodicTask {
-  private static final Logger LOGGER = LoggerFactory.getLogger(PeriodicTask.class);
-  protected ScheduledExecutorService _executorService;
-  private String _taskName;
+  private final String _taskName;
+  private long _executionTime;
   private long _intervalSeconds;
   private long _initialDelay;
 
+  public static final long DEFAULT_RUN_FREQUENCY_IN_SECOND = 3600L;
+  public static final long DEFAULT_INITIAL_DELAY = 120L;
+
   public PeriodicTask(String taskName) {
-    this(taskName, 3600L);
+    this(taskName, DEFAULT_RUN_FREQUENCY_IN_SECOND);
   }
 
   public PeriodicTask(String taskName, long runFrequencyInSeconds) {
-    this(taskName, runFrequencyInSeconds, 120L);
+    this(taskName, runFrequencyInSeconds, DEFAULT_INITIAL_DELAY);
   }
 
   public PeriodicTask(String taskName, long runFrequencyInSeconds, long initialDelay) {
     _taskName = taskName;
     _intervalSeconds = runFrequencyInSeconds;
     _initialDelay = initialDelay;
-    _executorService = Executors.newSingleThreadScheduledExecutor(runnable -> {
-      Thread thread = new Thread(runnable);
-      thread.setName(_taskName + "ExecutorService");
-      return thread;
-    });
+    _executionTime = System.currentTimeMillis() + initialDelay;
   }
 
   public abstract void runTask();
 
-  /**
-   * Start the periodic task.
-   */
-  public void start() {
-    LOGGER.info("Starting {}", _taskName);
-
-    // Set up an executor that executes validation tasks periodically
-    _executorService.scheduleWithFixedDelay(() -> {
-      try {
-        runTask();
-      } catch (Throwable e) {
-        // catch all errors to prevent subsequent exeuctions from being silently suppressed
-        // Ref: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay-java.lang.Runnable-long-long-java.util.concurrent.TimeUnit-
-        LOGGER.warn("Caught exception while running {}", _taskName, e);
-      }
-    }, _initialDelay, _intervalSeconds, TimeUnit.SECONDS);
-  }
-
-  /**
-   * Stop the periodic task.
-   */
-  public void stop() {
-    LOGGER.info("Stopping {}", _taskName);
-    if (_executorService == null) {
-      return;
-    }
-    _executorService.shutdown();
-    try {
-      _executorService.awaitTermination(_initialDelay, TimeUnit.SECONDS);
-    } catch (InterruptedException e) {
-      // Ignored
-    }
-    _executorService = null;
+  public void initTask() {
   }
 
   protected void setIntervalSeconds(long intervalSeconds) {
```

Comment:
That's because in `RealtimeSegmentRelocator`, we need to convert the frequency on the fly, e.g. converting `1h`(string format) to `3600`(long format). 
This conversion cannot be done until constructor gets called. And this setting intervalSeconds is what we're doing in the current code for `RealtimeSegmentRelocator`.